examples/src/rescue_raps/air.rs

```diff
@@ -0,0 +1,261 @@
+            &next[STATE_WIDTH..],
+            copy_flag,
+        );
+    }
+
+    fn evaluate_aux_transition<F, E>(
+        &self,
+        main_frame: &EvaluationFrame<F>,
+        aux_frame: &EvaluationFrame<E>,
+        periodic_values: &[F],
+        aux_rand_elements: &AuxTraceRandElements<E>,
+        result: &mut [E],
+    ) where
+        F: FieldElement<BaseField = Self::BaseField>,
+        E: FieldElement<BaseField = Self::BaseField> + ExtensionOf<F>,
+    {
+        let main_current = main_frame.current();
+        let main_next = main_frame.next();
+
+        let aux_current = aux_frame.current();
+        let aux_next = aux_frame.next();
+
+        let random_elements = aux_rand_elements.get_segment_elements(0);
+
+        let absorption_flag = periodic_values[1];
+
+        // We want to enforce that the absorbed values of the first hash chain are a
+        // permutation of the absorbed values of the second one. Because we want to
+        // copy two values per hash chain (namely the two capacity registers), we
+        // group them with random elements into a single cell via
+        // α_0 * c_0 + α_1 * c_1, where c_i is computed as next_i - current_i.
+
+        // Note that storing the copied values into two auxiliary columns. One could
+        // instead directly compute the permutation argument, hence require a single
+        // auxiliary one. For the sake of illustrating RAPs behaviour, we will store
+        // the computed values in additional columns.
+
+        let copied_value_1 = random_elements[0] * (main_next[0] - main_current[0]).into()
+            + random_elements[1] * (main_next[1] - main_current[1]).into();
+
+        result.agg_constraint(
+            0,
+            absorption_flag.into(),
+            are_equal(aux_current[0], copied_value_1),
+        );
+
+        let copied_value_2 = random_elements[0] * (main_next[4] - main_current[4]).into()
+            + random_elements[1] * (main_next[5] - main_current[5]).into();
+
+        result.agg_constraint(
+            1,
+            absorption_flag.into(),
+            are_equal(aux_current[1], copied_value_2),
+        );
+
+        // Enforce that the permutation argument column scales at each step by (aux[0] + γ) / (aux[1] + γ).
+        result.agg_constraint(
+            2,
+            E::ONE,
+            are_equal(
+                aux_next[2] * (aux_current[1] + random_elements[2]),
+                aux_current[2] * (aux_current[0] + random_elements[2]),
+            ),
+        );
+    }
+
+    fn get_assertions(&self) -> Vec<Assertion<Self::BaseField>> {
+        // Assert ending values of the hash chain
+        let last_step = self.trace_length() - 1;
+        vec![
+            Assertion::single(0, last_step, self.result[0][0]),
+            Assertion::single(1, last_step, self.result[0][1]),
+            Assertion::single(4, last_step, self.result[1][0]),
+            Assertion::single(5, last_step, self.result[1][1]),
+        ]
```

Comment:
We should probably enforce that columns 2, 3, and 6, 7 are set to zeros at the first step (at the beginning of the hash chain).

Reply:
Right, addressed!